builtin/providers/azurerm/encryption_settings.go

```diff
@@ -0,0 +1,163 @@
+package azurerm
+
+import (
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/disk"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func encryptionSettingsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"enabled": {
+					Type:     schema.TypeBool,
+					Required: true,
+
+					// Azure can change enabled from false to true, but not the other way around, so
+					//   to keep idempotency, we'll conservatively set this to ForceNew=true
+					// TODO: Is this the right behavior?
+					ForceNew: true,
+				},
+
+				"disk_encryption_key": {
+					Type:     schema.TypeList,
+					Optional: true,
```

Comment:
The current schema is:
  encryption_settings {
    enabled = true
    disk_encryption_key { ... }
    key_encryption_key { ... }
  }

If enabled=true at least DEK must be supplied, but I can't mark it required because if enabled=false, DEK and KEK are not allowed to be set by Azure.  Would it be better to not expose enabled and just infer it as true if encryption_settings is supposed?